### PR TITLE
Ct zone

### PR DESCRIPTION
**Pull request by elibritstein**
i see those commits reveal a bug - a crash in stopping the OVS.
the crash is in netdev_dpdk_get_hw_id.
keeping the OVS up, i don't see a problem (probably hidden).
i issue this PR anyway, for review and for Baidu

### CODE REVIEW

File: lib/netdev-rte-offloads.c [hunk 1]
```diff
@@ -511,11 +511,43 @@ add_flow_rss_action(struct flow_actions *actions,
...
+    TAG_FIELD_NUM,
+};
+
+/* map tag_id to register index, mask as few tags can share the same register
+ * and data field is used as the bit shift needed.
+ */
+static struct rte_flow_action_set_tag tag_id_maps[] = {
+    [TAG_FIELD_HW_ID] = {
+        .index = 0,
+        .mask = 0xFFFFFFFF,
```
Comment (rbaryanai) [inline lib/netdev-rte-offloads.c]: HW id should be 16 bits

File: lib/netdev-rte-offloads.c [hunk 2]
```diff
@@ -511,11 +511,43 @@ add_flow_rss_action(struct flow_actions *actions,
...
+        .data = 0,
+    },
+    [TAG_FIELD_CT_STATE] = {
+        .index = 1,
+        .mask = 0x000000FF,
+        .data = 1,
```
Comment (rbaryanai) [inline lib/netdev-rte-offloads.c]: should not it be zero shift?

File: lib/netdev-rte-offloads.c [hunk 3]
```diff
@@ -511,11 +511,43 @@ add_flow_rss_action(struct flow_actions *actions,
...
+    TAG_FIELD_CT_MARK,
+    TAG_FIELD_NUM,
+};
+
+/* map tag_id to register index, mask as few tags can share the same register
+ * and data field is used as the bit shift needed.
+ */
+static struct rte_flow_action_set_tag tag_id_maps[] = {
+    [TAG_FIELD_HW_ID] = {
+        .index = 0,
```
Comment (rbaryanai) [inline lib/netdev-rte-offloads.c]: index 0/1 are reserved for meta and mark and shouldn't be used. I know this is a PMD issue but do they handle it?

File: lib/netdev-rte-offloads.c (hunk 1 above)
Comment (elibritstein) [reply lib/netdev-rte-offloads.c]: hw id should not be in a tag at all. will be in metadata (in mlx case 16 bit)

File: lib/netdev-rte-offloads.c (hunk 3 above)
Comment (elibritstein) [reply lib/netdev-rte-offloads.c]: it is different API in DPDK. the app doesn't know about it, and can use index 0/1 too

Comment (rbaryanai) [reply lib/netdev-rte-offloads.c]: great

File: lib/netdev-rte-offloads.c (hunk 1 above)
Comment (rbaryanai) [reply lib/netdev-rte-offloads.c]: So why is it here?

Comment (elibritstein) [reply lib/netdev-rte-offloads.c]: because it is not the in the scope of those commits

File: lib/netdev-rte-offloads.c (hunk 2 above)
Comment (elibritstein) [reply lib/netdev-rte-offloads.c]: fixed

**Pull request merged**